include/simdjson/generic/ondemand/object-inl.h

```diff
@@ -18,12 +18,23 @@ namespace ondemand {
 
 #ifdef SIMDJSON_SUPPORTS_EXTRACT
 
+
+#ifdef _MSC_VER
 template <endpoint ...Funcs>
-simdjson_inline error_code object::extract(Funcs&&... endpoints)
-#ifndef _MSC_VER // msvc thinks noexcept is not the same in definition
- noexcept((nothrow_endpoint<Funcs> && ...))
-#endif
-{
+simdjson_inline error_code object::extract(Funcs&&... endpoints) {
+  return iter.on_field_raw([&, eps = std::make_tuple(std::forward<Funcs>(endpoints)...)](auto field_key, error_code& error) mutable {
+    std::apply([&](auto &...endpoints) {
+       std::ignore = ((field_key.unsafe_is_equal(endpoints.key()) ? (error = endpoints(value(iter.child()))) == SUCCESS : true) && ...);
```

Comment:
I have no idea why MSVC gives that error; I fixed it the way programmers fix everything, adding another layer of indirection.

I could not see any visible performance drop after this though.

Reply:
Please see https://github.com/the-moisrex/simdjson-custom-types/pull/2 for a slight tweak.